### PR TITLE
Add sender report trace event

### PR DESCRIPTION
**Pull request by GithubUser8080**
This PR adds a trace event for incoming RTCP sender reports from producers, to provide timestamp information in the node API. Covers use cases of server-side stream synchronization (like recording).

Usage 

`producer.enableTraceEvent(['sr'])`

emits

```
{
  type: 'sr',
  timestamp: 6348927,
  direction: 'in',
  info: {
    ssrc: 481847794,
    ntpSec: 3911629480,
    ntpFrac: 2788138149,
    rtpTs: 2712133695,
    packetCount: 928,
    octetCount: 75009
  }
}
```

Resolves issue https://github.com/versatica/mediasoup/issues/1248

### CODE REVIEW

**Review by ibc — CHANGES_REQUESTED** [review]
Looks good but Rust changes are also required.

**Comment by GithubUser8080** [comment]
Not familiar with rust so I am relying on the videoroom example for manual testing, hope it is enough.

**Comment by ibc** [comment]
Just do `grep` looking for similar FBS types in worker/rust folder and will be easy.

**Comment by ibc** [comment]
Run  `cargo fmt` to fix lint/clippy formating issues.

File: worker/fbs/producer.fbs [hunk 1]
```diff
@@ -60,6 +61,7 @@ union TraceInfo {
     FirTraceInfo,
     PliTraceInfo,
     RtpTraceInfo,
+    SRTraceInfo,
```
Comment (jmillan) [inline worker/fbs/producer.fbs]: For consistency,

```suggestion
    SrTraceInfo,
```

File: worker/fbs/producer.fbs [hunk 2]
```diff
@@ -80,6 +82,15 @@ table RtpTraceInfo {
     is_rtx: bool;
 }
 
+table SRTraceInfo {
```
Comment (jmillan) [inline worker/fbs/producer.fbs]:
```suggestion
table SrTraceInfo {
```

File: worker/include/RTC/Producer.hpp [hunk 3]
```diff
@@ -160,6 +161,7 @@ namespace RTC
 		void EmitTraceEventPliType(uint32_t ssrc) const;
 		void EmitTraceEventFirType(uint32_t ssrc) const;
 		void EmitTraceEventNackType() const;
+		void EmitTraceEventSRType(RTC::RTCP::SenderReport* report) const;
```
Comment (jmillan) [inline worker/include/RTC/Producer.hpp]:
```suggestion
		void EmitTraceEventSrType(RTC::RTCP::SenderReport* report) const;
```

File: worker/src/RTC/Producer.cpp [hunk 4]
```diff
@@ -706,6 +712,8 @@ namespace RTC
 
 			this->listener->OnProducerRtcpSenderReport(this, rtpStream, first);
 
+			EmitTraceEventSRType(report);
```
Comment (jmillan) [inline worker/src/RTC/Producer.cpp]:
```suggestion
			EmitTraceEventSrType(report);
```

**Review by jmillan — COMMENTED** [review]
Looks good so far :+1:, just a cosmetic naming change requested.

**Review by jmillan — APPROVED** [review]

**Review by ibc — APPROVED** [review]

**Comment by ibc** [comment]
Changelog entry missing and this must be documented in the website (I'll do).

File: rust/src/data_structures.rs [hunk 5]
```diff
@@ -1234,3 +1236,34 @@ impl BweTraceInfo {
         }
     }
 }
+
+/// BWE info in trace event.
```
Comment (ibc) [inline rust/src/data_structures.rs:1268]: This is not "BWE info".

**Review by ibc — CHANGES_REQUESTED** [review]

File: rust/src/data_structures.rs (hunk 5 above)
Comment (ibc) [reply rust/src/data_structures.rs:1268]: I'll merge and make this change in v3.

**Comment by GithubUser8080** [comment]
Thank you

**Pull request merged**